### ISSUE TITLE
Set authentication credentials for requests from Titanium

### ISSUE THREAD

**Issue by melanchall** [justcoding121/Titanium-Web-Proxy#800]
Hi,

Is it possible to set authentication credentials that will be used with each request? To understand what I mean, look please at Fiddler and its ability to specify `X-AutoAuth` parameter: https://docs.telerik.com/fiddler/Configure-Fiddler/Tasks/AuthenticateWithCBT. From the mentioned article:

> To use creds other than your Windows login credentials, set X-AutoAuth to "domain\\username:password"

Is it possible to specify desired creds with Titanium?

**Comment by honfika** [comment]
You can do it manually in the BeforeResponse event if you know how to set this authentication headers for windows. (I don't know)

**Comment by melanchall** [comment]
I need to set custom credentials for NTLM/Negotiate authentication. From [the docs](https://justcoding121.github.io/Titanium-Web-Proxy/docs/api/Titanium.Web.Proxy.ProxyServer.html#Titanium_Web_Proxy_ProxyServer_EnableWinAuth) I see:

> EnableWinAuth
> Enable disable Windows Authentication (NTLM/Kerberos). Note: NTLM/Kerberos will always send local credentials of current user running the proxy process. This is because a man in middle attack with Windows domain authentication is not currently supported. Defaults to false.

So right now there is no way to set arbitrary credentials with Titanium?

**Comment by honfika** [comment]
It is not possbile with an easy property setting.
You can do it only manually.

**Comment by melanchall** [comment]
Are there any plans to provide simple way of doing it like Fiddler provides?

**Comment by honfika** [comment]
No, sorry... but we are accepting pull request.. so if you implement it, it will be included:)

**Comment by justcoding121** [comment]
Closing with guidance after the #461 credential-provider work:

- **NTLM/Negotiate/Kerberos (WinAuth):** use `ProxyServer.EnableWinAuth` + `WinAuthCredentialsProvider` (see #461 / 6b3e8c2d). Do not put passwords on session args.
- **HTTP Basic:** set the `Authorization` header yourself in `BeforeRequest` (e.g. `Basic base64(user:pass)`), or rely on upstream-proxy Basic fields on `ExternalProxy`. There is no Fiddler-style `X-AutoAuth` header interpreter built in.

**Comment by justcoding121** [comment]
Closing: use WinAuthCredentialsProvider (#461) for NTLM/Negotiate, or BeforeRequest for Basic — see previous comment.
